packages/next/src/server/app-render/app-render.tsx

```diff
@@ -3124,19 +3127,66 @@ async function renderWithRestartOnCacheMissInDev(
           }
         )
       },
-      (stream) => {
-        // Runtime stage
-        finalStageController.advanceStage(RenderStage.Runtime)
-        return stream
+      async (stream) => {
+        const [continuationStream, staticStream] = stream.tee()
```

Comment:
did you benchmark this

this would likely only be faster if you can move it one layer up, to access the chunks before they get pushed

otherwise this will likely create a large number of promises that make it slower